core/fields/Field_Rich.php

```diff
@@ -201,9 +201,77 @@ public function display() {
 						document.execCommand('formatBlock', false, command);
 					}
 					
+					// note that 'insertimage' command is not available in text editor.
 					else if (command == 'createlink' || command == 'insertimage') {
-						url = prompt('Enter the link here: ','https:\/\/');
-						document.execCommand(command, false, url);
+						
+						// get selected text to add to link
+						var selection = window.getSelection().toString();
+
+						// insert html at anchor location
+						var link_html = "<a id='newly_created_link'>" + selection + "</a>";
+						document.execCommand('insertHTML', false, link_html);							link = document.getElementById('newly_created_link');
+						link.removeAttribute('style');	// get rid of any styling that might be preserved from user copy/paste
+
+						// display modal
+						var url = "";
+						var modal = document.createElement('div');
+						modal.innerHTML = `
+							<div class="modal-background"></div>
+							<div class="modal-content">
+								<div class="box">
+
+									<div class="field">
+										<label class="label">Link Address</label>
+										<div class="control">
+											<input id="created_url" class="input" type="text" value="` + url + `">
+										</div>
+										<p class='help'>Enter full link including https://</p>
+									</div>
+									
+									<div class="field">
+										<label class="label">Display Text</label>
+										<div class="control">
+											<input id="text_to_display" class="input" placeholder='' type="text" value="` + selection + `">
+										</div>
+									</div>
+
+									<button id="modal_save" class="button is-primary">Add</button>
+									<button id="modal_cancel" class="button is-warning">Cancel</button>
+
+								</div>
+							</div>
+						`;
+						modal.classList = "modal";
+						document.body.appendChild(modal);
+						modal.classList.add("is-active");
```

Comment:
overwrite, and the an add, might be better to do it all on one line

Reply:
Done.